packages/test/test-end-to-end-tests/src/test/summaryDataStoreStats.spec.ts.ts

```diff
@@ -0,0 +1,179 @@
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { TelemetryNullLogger } from "@fluidframework/common-utils";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { IContainer } from "@fluidframework/container-definitions";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { MockLogger } from "@fluidframework/test-runtime-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { IContainerRuntimeOptions, SummaryCollection } from "@fluidframework/container-runtime";
+import { flattenRuntimeOptions } from "./flattenRuntimeOptions";
+
+const _global: any = global;
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+
+    public get _runtime() {
+        return this.runtime;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+}
+
+describeNoCompat("Generate Summary Stats", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    const dataObjectFactory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const IdleDetectionTime = 100;
+    const summaryConfigOverrides: Partial<ISummaryConfiguration> = {
+        idleTime: IdleDetectionTime,
+        maxTime: IdleDetectionTime * 12,
+    };
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            generateSummaries: true,
+            initialSummarizerDelayMs: 10,
+            summaryConfigOverrides,
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        undefined,
+        flattenRuntimeOptions(runtimeOptions),
+    );
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+    let summaryCollection: SummaryCollection;
+
+    /**
+     * Waits for a summary with the current state of the document (including all in-flight changes). It basically
+     * synchronizes all containers and waits for a summary that contains the last processed sequence number.
+     * @returns the sequence number of the summary
+     */
+     async function waitForSummary(): Promise<number> {
+        await provider.ensureSynchronized();
+        const sequenceNumber = mainContainer.deltaManager.lastSequenceNumber;
+        await summaryCollection.waitSummaryAck(sequenceNumber);
+        return sequenceNumber;
+    }
+
+    function getGenerateSummaryEvent(sequenceNumber: number): ITelemetryBaseEvent | undefined {
+        const events = mockLogger.events;
```

Comment:
nit: you don't need this. you can do `for (const event of mockLogger.events)` below.